piv: block use of RSA1024 keys on YubiKey FIPS

dagheyman opened the pull request.

dagheyman commented:
(not ready yet)

emlun commented on ykman/cli/piv.py:
```diff
@@ -792,10 +794,17 @@ def _authenticate(ctx, controller, management_key, mgm_key_prompt,
+
+def _check_key_size(ctx, private_key):
+    if ctx.obj['dev'].is_fips and private_key.key_size == 1024:
+        ctx.fail('1024 is not a supoported key size on this YubiKey.')
```
Typo: supoported -> supported

emlun commented on ykman/cli/piv.py:
```diff
@@ -792,10 +794,17 @@ def _authenticate(ctx, controller, management_key, mgm_key_prompt,
     if algorithm == ALGO.ECCP384 and controller.version < (4, 0, 0):
         ctx.fail('ECCP384 is not supported by this YubiKey.')
+    if algorithm == ALGO.RSA1024 and ctx.obj['dev'].is_fips:
+        ctx.fail('RSA1024 is not supoported by this YubiKey.')
```
Typo: supoported -> supported

dagheyman commented:
Added a couple of tests and fixed the typo.

> On second thought, shouldn't this be done in the library instead of the CLI?

Maybe, but it fitted well next to the other checks in `cli/piv.py`. Let's discuss that more and maybe do separate refactoring PRs to move these kind of checks to the library level, if that is what we want.

emlun commented:
Sounds good.
